appserver/admingui/payara-console-extras/src/main/java/fish/payara/admingui/extras/rest/PayaraRestApiHandlers.java

```diff
@@ -444,5 +444,98 @@ public static void getHistoricHealthcheckMessages(HandlerContext handlerCtx){
+        String configName = (String) handlerCtx.getInputValue("configName");
+        String sessionScopeRestURL = (String) handlerCtx.getInputValue("sessionScopeRestURL");
+        sessionScopeRestURL = sessionScopeRestURL.endsWith("/") ? sessionScopeRestURL : sessionScopeRestURL + "/";
+        String serverName = "";
        
+        try {
+            List<Map> table = RestUtil.buildChildEntityList(
+                    (String)handlerCtx.getInputValue("parentEndpoint"),
+                    (String)handlerCtx.getInputValue("childType"),
+                    (List)handlerCtx.getInputValue("skipList"),
+                    (List)handlerCtx.getInputValue("includeList"),
+                    (String)handlerCtx.getInputValue("id"));
+            
+            if (configName.equals("default-config")) {
+                for (Map row : table) {
+                    row.put("contextRoot", "");
+                }
+            } else {
+                List<String> instances = RestUtil.getChildList(sessionScopeRestURL + "servers/server");
+                for (String instance : instances) {
+                    String configRef = (String) RestUtil.getAttributesMap(instance).get("configRef");
+                    if (configRef.equals(configName)) {
+                        serverName = instance.substring(instance.lastIndexOf("/") + 1);
+                    }
+                }
+                String deployedApplicationsEndpoint = sessionScopeRestURL + "servers/server/" + serverName 
+                        + "/application-ref";
+
+                List<String> deployedApplications = RestUtil.getChildList(deployedApplicationsEndpoint);
+                List<String> virtualServers = RestUtil.getChildList(parentEndpoint + "/" + childType);
+                List<String> applications = RestUtil.getChildList(sessionScopeRestURL + "applications/application");
+
+                for (String virtualServer : virtualServers) {         
+                    String virtualServerName = virtualServer.substring(virtualServer.lastIndexOf("/") + 1);
+
+                    for (int i = 0; i < deployedApplications.size(); i++) {
+                        deployedApplications.set(i, deployedApplications.get(i)
+                                .substring(deployedApplications.get(i).lastIndexOf("/") + 1));
+                    }
+
+                    String contextRoots = "";
+
+                    for (String application : applications) {
+                        String applicationName = application.substring(application.lastIndexOf("/") + 1);
+                        String[] deployedVirtualServers; 
+                        if (RestUtil.get(deployedApplicationsEndpoint + "/" + applicationName).isSuccess()) {
+                            deployedVirtualServers = ((String) RestUtil.getAttributesMap(
+                                    deployedApplicationsEndpoint + "/" + applicationName).get("virtualServers")).split(",");
```

Comment:
If the application is deployed to the blank virtual server, this will throw an error because the virtual server list is null